shopinvader_locomotive_impersonate/services/customer.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _
+from odoo.exceptions import UserError
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.component.core import Component
+
+
+class CustomerService(Component):
+    _inherit = "shopinvader.customer.service"
+
+    @restapi.method(
+        routes=[(["/impersonate"], "GET")],
+        input_param=restapi.CerberusValidator("_validator_impersonate"),
+    )
+    def impersonate(self, email=None, token=None):
+        shop_partner = self.env["shopinvader.partner"]._get_from_token(email, token)
+        if shop_partner:
+            customer = self._to_customer_info(shop_partner.record_id)
+            return {
+                "store_cache": {"customer": customer},
+                "force_authenticate_customer": shop_partner.email,
+                "redirect_to": shop_partner.backend_id.location,
+            }
+        else:
+            raise UserError(_("Invalid impersonate token"))
```

Comment:
```suggestion
            raise AccessError(_("Invalid impersonate token"))
```
maybe it's more accurate?